### DISCUSSION TITLE
PeerAddr relies on local configuration to dial a peer

### DISCUSSION THREAD

**Discussion by dvc94ch** [n0-computer/iroh#1571] [General]
The region id is local configuration information. For iroh-net to work all peers need to use the same derp map configuration.

**Comment by dignifiedquire** [comment]
Yes, this is by design currently, so that nodes can decide which derp servers to use and which not. (Not saying this is optimal, just that the current design is intended to work like this) 

> need to use the same derp map configuration.

Only the regions that are being used have to overlap, not the whole configuration

**Comment by dvc94ch** [comment]
Can you elaborate? Which derp servers to use for incoming traffic each peer decides. But for outgoing traffic you need to know what the other peer decided

**Reply by dignifiedquire** [reply]
> But for outgoing traffic you need to know what the other peer decided

Yes, but you only need to know this for peers you actually communicate with.

**Comment by dvc94ch** [comment]
It would be really cool if iroh-net turned into something suitable for p2p networking in general. I think that this needs to be resolved for it to be usable by non iroh networks.

**Reply by dignifiedquire** [reply]
What is the stopper here? You can choose your own set of derp regions + nodes, and use them for every peer in your network. This can be done completely independent of the rest of iroh.

**Reply by eminence** [reply]
I feel like most iroh-net use cases will probably just use a single global derp server, so I'm also not sure there is any blocker here

**Reply by dvc94ch** [reply]
a single global derp server  is not going to satisfy all users, due to bandwidth, latency and availability concerns. if you're deploying iroh in an intranet, sure you can have a single derp server for your companies intranet.

each iroh node picks it's derp servers and each iroh node that wants to dial another iroh node has no way of knowing what derp servers it picked.

**Reply by dvc94ch** [reply]
that is assuming the global derp server is not malicious and arbitrarily censors some peers due to the fbi wanting to suppress the hunter biden story during the election or other kinds of election interference that despite overwhelming evidence, the democrats still claim the election wasn't stolen.

**Reply by dvc94ch** [reply]
canada is trying to suppress the roe rogan podcast through new legislation. if you think a single global derp server is going to prevent governments from censoring news worthy stories, you're a bit naive

**Reply by dvc94ch** [reply]
facebook and instagram are blocked in canada now because of new legislation that meta didn't want to comply with

**Reply by dignifiedquire** [reply]
derp servers can't censor any data, as the connections are end to end encrypted. The only thing they could choose to is not allow certain peerids (easily fixed by using rotating ids) or IPs.

**Reply by dvc94ch** [reply]
This means a peerid can't have any reputation and the source can't be verified or trusted because you need to change it all the time. This means you can't trust anything and spreading verified and true stories is the same as spreading anonymous unverified propaganda.

I assume what you're proposing is building a stable long term signing key system on top and a discovery system that can map the long term key to an ephemeral one.

**Reply by dvc94ch** [reply]
Well if there is an automated way to find the short term peerid you can still censor, so the problem isn't really solved. Also neglects that they can outlaw the system and take it all down if there is just one derper. (An example of that would be trying to outlaw blockchain)

**Comment by rklaehn** [comment]
@dvc94ch just so you know, everybody at n0 cares deeply about enabling p2p applications. You can take that as a given.

We will deviate from p2p purity if we think the pure p2p solution will not work reliably or have bad performance. We want iroh to be competitive with web2.

For projects that place p2p purity above performance and reliability concerns, we provide plenty of very nice open source building blocks.

Now that you know this, can you please leave politics out of technical discussions in the future? It is not helpful at all.
